torch/_dynamo/variables/tensor.py

```diff
@@ -681,36 +684,59 @@ class TensorWithTFOverrideVariable(VariableTracker):
     def create(
         tx,
         tensor_variable,
-        orig_tensor_variable_source,
+        tensor_variable_source,
         torch_function_fn,
         subclass_type,
         **kwargs,
     ):
         var = TensorWithTFOverrideVariable(
             tensor_variable,
-            orig_tensor_variable_source,
+            tensor_variable_source,
             torch_function_fn,
             subclass_type,
             **kwargs,
         )
         # stash the subclass type to rewrap an output tensor if needed
-        tx.output.install_global(var.global_class_name(), subclass_type)
+        if var.global_class_name() not in tx.output.global_scope:
+            tx.output.install_global(var.global_class_name(), subclass_type)
```

Comment:
Why do we have to install globals for subclasses?

Reply:
I had to install a global so that we could properly "rewrap" the tensor subclass if we return it. This to get around not fake-ifying the subclass directly. Some python bytecode is inserted during codegen to rewrap the output if it was a subclass.